website/content/en/docs/best-practices/common-recommendation.md

```diff
@@ -0,0 +1,73 @@
+---
+title: "Common recommendations and suggestions"
+linkTitle: "Common suggestions"
+weight: 2
+description: Common recommendations and suggestions to built solutions with Operator SDK
+---
+
+## Overview
+
+Any recommendations or good practices suggested by the Kubernetes community, such as how to to [develop Operator pattern solutions][operator-pattern] or how to [use controller-runtime][controller-runtime] are good recommendations for those who are looking to build operator projects with operator-sdk. Also, see [Operator Best Practices][operator-best-practices]. However, here are some common recommendations.
+
+## Common Recommendations
+
+### Develop idempotent reconciliation solutions
+
+When developing operators, it is important for the controller's reconciliation loop to be idempotent. By following the [Operator pattern][operator-pattern] you will create [Controllers](https://kubernetes.io/docs/concepts/architecture/controller/) which provide a reconcile function responsible for synchronizing resources until the desired state is reached on the cluster.
+
+### Understanding Kubernetes APIs
+
+Building your own operator commonly involves extending the Kubernetes API itself. It is helpful to understand exactly how [Custom Resource Definitions interact with the Kubernetes API][k8s-crd-doc]. Also, the Kubebuilder documentation on [Groups and Versions and Kinds] may be helpful to better understand these concepts as they relate to operators.
+
+### Avoid a design solution where more than one Kind is reconciled by the same controller
+
+Having many Kinds (such as CRDs) which are all managed by the same controller usually goes against the design proposed by [controller-runtime.][controller-runtime]. Furthermore this might hurt concepts such as encapsulation, the Single Responsibility Principle, and Cohesion. Damaging these concepts may cause unexpected side effects, and increase the difficulty of extending, reusing, or maintaining the operator.
+
+### Others common suggestions
+
+Additionally, we can share:
+
+- Provide the images and tags used by the operator solution via environment variables in the `config/manager/manager.yaml`: 
+
+```yaml
+...
+spec:
+  ...
+    spec:
+      ...
+      containers:
+      - command:
+        - /manager
+        ...
+        env:
+        - name: MY_IMAGE
+          value: "quay.io/example.com/image:0.0.1"
+```
+
+- Manage your solutions using [Status Conditionals][status-conditionals] 
+- Use [finalizers][finalizers] when/if required 
+- Cover the project with tests/CI to ensure its quality:
+    - For any language-based operator, you can use [Scorecard][scorecard] to implement functional tests
+    - For Go-based operators, you can also use [envtest][envtest] to cover the controllers. For further information see [Testing with EnvTest][testing-with-envtest]
+    - For Ansible-based operators, you can also use [Molecule][molecule]. For further information see [Testing with Molecule][molecule-tests]
```

Comment:
i would put a blurb about what molecule is. like
"you can also use [Molecule][molecule], an Ansible testing framework."